backend/api/tests/test_StreamPosts.py

```diff
@@ -0,0 +1,427 @@
+                "unlisted": False
+                }
+
+        self.server_only_post = {
+                "title": "server_only_post title",
+                "source": "http://lastplaceigotthisfrom.com/posts/yyyyy",
+                "origin": "http://whereitcamefrom.com/posts/zzzzz",
+                "description": "server_only_post description",
+                "contentType": "text/plain",
+                "content": "server_only_post content",
+                "categories": ["test_category_1", "test_category_2"],
+                "published": "2015-03-09T13:07:04+00:00",
+                "id": "de305d54-75b4-431b-adb2-eb6b9e546013",
+                "visibility": "SERVERONLY",
+                "visibleTo": [],
+                "unlisted": False
+                            }
+
+    def test_invalid_auth(self):
+        response = self.client.get("/api/author/posts")
+        self.assertEqual(response.status_code, 403)
+
+
+    def test_get_own_posts_in_stream(self):
+        Post.objects.all().delete()
+        Follow.objects.all().delete()
+        self.client.login(username=self.username1, password=self.password1)
+
+        create_mock_post(self.public_post1, self.authorProfile1)
+        create_mock_post(self.public_post2, self.authorProfile1)
+
+        response = self.client.get("/api/author/posts")
+
+        expected_output = {
+            "query": "posts",
+            "count": 2,
+            "posts": [self.public_post2, self.public_post1]
+        }
+
+        expected_author_list = [self.authorProfile1, self.authorProfile1] 
+
+        self.assertEqual(response.status_code, 200)
+        assert_post_response(response, expected_output, expected_author_list)
+        self.client.logout()
+
+    def test_get_public_post_following_others(self):
+        Post.objects.all().delete()
+        Follow.objects.all().delete()
+        self.client.login(username=self.username1, password=self.password1)
+
+        Follow.objects.create(
+                authorA=self.user_id_1,
+                authorB=self.user_id_2,
+                status="FOLLOWING")
+        
+        create_mock_post(self.public_post1, self.authorProfile1)
+        create_mock_post(self.public_post2, self.authorProfile2)
+        create_mock_post(self.public_post2, self.authorProfile2)
+
+        response = self.client.get("/api/author/posts")
+
+        expected_output = {
+            "query": "posts",
+            "count": 3,
+            "posts": [self.public_post2, self.public_post2, self.public_post1]
+        }
+
+        expected_author_list = [self.authorProfile2, self.authorProfile2, self.authorProfile1] 
+
+        self.assertEqual(response.status_code, 200)
+        assert_post_response(response, expected_output, expected_author_list)
+        self.client.logout()
+
+    
+    def test_get_unlisted_post(self):
+        Post.objects.all().delete()
+        self.client.login(username=self.username1, password=self.password1)
+
```

Comment:
make sure the follow objects is cleared by doing `Follow.objects.all().delete() ` for each test to avoid test interfering with each other